featuretools/entityset/entity.py

```diff
@@ -365,12 +370,16 @@ def infer_variable_types(self, ignore=None, link_vars=None):
                         inferred_type = vtypes.Datetime
                     else:
                         # heuristics to predict this some other than categorical
-                        sample = df[variable].sample(min(10000, df[variable].nunique()))
-                        avg_length = sample.str.len().mean()
-                        if avg_length > 50:
-                            inferred_type = vtypes.Text
+                        nunique = df[variable].dropna().nunique()
+                        if nunique == 2:
+                            inferred_type = vtypes.Boolean
                         else:
-                            inferred_type = vtypes.Categorical
+                            sample = df[variable].sample(min(10000, nunique))
```

Comment:
can you explain why you do `min(10000, nunique)`? Say, `nunique == 5`, wouldn't that be too small of a sample?

Reply:
no because that would be all of the values... so not a sample. However, if nunique is that small then we should just default to Categorical over Text. So I think there should be another conditional for nunique < some heuristic